### PR TITLE
mapbox-wagyu: add recipe

### PR DESCRIPTION
**Pull request by toge**
Specify library name and version:  **mapbox-wagyu/0.5.0**

Try to close https://github.com/conan-io/conan-center-index/issues/22579.

---

- [x] I've read the [contributing guidelines](https://github.com/conan-io/conan-center-index/blob/master/CONTRIBUTING.md).
- [x] I've used a [recent](https://github.com/conan-io/conan/releases/latest) Conan client version close to the [currently deployed](https://github.com/conan-io/conan-center-index/blob/master/.c3i/config_v1.yml#L6).
- [x] I've tried at least one configuration locally with the [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

**Review by jellespijker — APPROVED** [review]
lgtm

**Review by casperlamboo — APPROVED** [review]

File: recipes/mapbox-wagyu/all/conanfile.py [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+from conan.tools.layout import basic_layout
+from conan.tools.scm import Version
+import os
+
+required_conan_version = ">=1.52.0"
+
+class MapboxWagyuConan(ConanFile):
+    name = "mapbox-wagyu"
+    description = "A general library for geometry operations of union, intersections, difference, and xor"
+    license = "BSD-2-Clause"
```
Comment (uilianries) [inline recipes/mapbox-wagyu/all/conanfile.py:14]:
```suggestion
    license = "LicenseRef-mapbox-wagyu"
```

The license is not a standard open source listed in SPDX. It contains parts of MIT, but the second paragraph is customized. 

Plus, in https://github.com/mapbox/wagyu/blob/master/include/mapbox/geometry/wagyu/almost_equal.hpp it has a BSD-3 customized too.

File: recipes/mapbox-wagyu/all/conanfile.py [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+            "gcc": "7",
+            "msvc": "191",
+            "Visual Studio": "15",
+        }
+
+    def layout(self):
+        basic_layout(self, src_folder="src")
+
+    def requirements(self):
+        self.requires("boost/1.84.0", transitive_headers=True)
```
Comment (uilianries) [inline recipes/mapbox-wagyu/all/conanfile.py:40]:
```suggestion
```
I don't see boost being consumed by the project headers, only benchmark and tests. Am I missing something? https://github.com/search?q=repo%3Amapbox%2Fwagyu%20boost&type=code

File: recipes/mapbox-wagyu/all/conanfile.py [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+            "gcc": "7",
+            "msvc": "191",
+            "Visual Studio": "15",
+        }
+
+    def layout(self):
+        basic_layout(self, src_folder="src")
+
+    def requirements(self):
+        self.requires("mapbox-geometry/2.0.3", transitive_headers=True)
```
Comment (RubenRBS) [inline recipes/mapbox-wagyu/all/conanfile.py:40]: fyi: `package_type = "header-library"` imples `transitive_headers=True` for every requirement of the recipe :)

**Review by RubenRBS — COMMENTED** [review]
One tip, otherwise looks good - not approving to let @uilianries have a last say before merge, but looks good to me otherwise :)

File: recipes/mapbox-wagyu/all/conanfile.py (hunk 3 above)
Comment (toge) [reply recipes/mapbox-wagyu/all/conanfile.py]: @RubenRBS
Thanks a lot!
I removed transitive_headers.

**Review by RubenRBS — APPROVED** [review]

**Review by juansblanco — APPROVED** [review]

**Pull request merged**